transformer/classes/dockerfile/rubydockerfilegenerator.go

```diff
@@ -0,0 +1,165 @@
+/*
+ *  Copyright IBM Corporation 2021
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dockerfile
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/konveyor/move2kube/common"
+	"github.com/konveyor/move2kube/environment"
+	"github.com/konveyor/move2kube/types/qaengine/commonqa"
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+// RubyDockerfileGenerator implements the Transformer interface
+type RubyDockerfileGenerator struct {
+	Config       transformertypes.Transformer
+	Env          *environment.Environment
+	NodejsConfig NodejsDockerfileYamlConfig
+}
+
+// RubyTemplateConfig implements Nodejs config interface
+type RubyTemplateConfig struct {
+	Port    int32
+	AppName string
+}
+
+// Init Initializes the transformer
+func (t *RubyDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *RubyDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *RubyDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *RubyDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	Gemfiles, err := common.GetFilesByName(dir, []string{"", "Gemfile"}, nil)
+	if err != nil {
+		logrus.Debugf("Cannot get the Gemfile: %s", err)
+		return nil, nil, nil
+	}
+	if len(Gemfiles) > 0 {
+		rubyFiles, err := common.GetFilesByExt(dir, []string{".rb"})
+		if err != nil {
+			logrus.Errorf("Error while finding ruby files %s", err)
+		}
+		var serviceName string
+		if len(rubyFiles) == 1 {
+			serviceName = strings.TrimSuffix(filepath.Base(rubyFiles[0]), ".rb")
+		}
+		if serviceName != "" {
+			namedServices = map[string]transformertypes.ServicePlan{
+				serviceName: []transformertypes.TransformerPlan{{
+					Mode:              t.Config.Spec.Mode,
```

Comment:
Mughe be better to create the transformer plan once and use it in both the places depending on whether it is named or unnamed

Reply:
Yes sure, done.